### ISSUE TITLE
Emulate FOR UPDATE .. WAIT n using SET LOCAL LOCK_TIMEOUT in PostgreSQL dialects

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#11243]
In PostgreSQL and similar dialects, there's a `SET LOCAL LOCK_TIMEOUT` flag that can be set to achieve the same effect as in Oracle's `FOR UPDATE .. WAIT <n seconds>` syntax. See:
https://stackoverflow.com/q/20963450/521799

We could emulate `WAIT` using a compound statement that sets the timeout and resets it to the previous value after the statement.

### Caveat: First result is update count

When we run a statement batch like this:

```sql
SET LOCAL LOCK_TIMEOUT = x;
SELECT .. FOR UPDATE;
```

Then, the first result is the update count of `0`, which can be seen when fetching the above with `DSLContext.fetchMany()`. jOOQ can skip that first result, but clients may have to do this manually. I don't see any way around this, as e.g. `DO` statements can't return result sets.

### Caveat: Resetting the `LOCK_TIMEOUT`

We should retrieve the previous value using `SHOW LOCK_TIMEOUT` and reset it after executing the query. But `DO` cannot return result sets, so the workaround would be extremely laborious:

```sql
do $$
  declare
    _c1 CONSTANT refcursor := '_c1';
    _t interval;
  begin
    show lock_timeout into _t;
    execute 'set local jooq.previous_lock_timeout = ''' || (extract(epoch from _t) * 1000) || '''';
    set local lock_timeout = 1000;
    open _c1 for select * from t_book for update /* wait 1 emulation */;
  end
$$;
fetch all from _c1;
close _c1;
do $$ 
  declare
    _t int;
  begin
    show jooq.previous_lock_timeout into _t;
    execute 'set local lock_timeout = ''' || _t || '''';
    reset jooq.previous_lock_timeout;
  end
$$;
```

I'm currently not too keen on implementing this, until someone requests it, or until we have more use-cases where we need to set and reset some local parameter like `lock_timeout`.

**Comment by lukaseder** [comment]
Fixed for jOOQ 3.15.0

**Comment by lukaseder** [comment]
MySQL version: https://github.com/jOOQ/jOOQ/issues/11543
